### PR TITLE
feat: return multiple devfiles when having multiple components

### LINKED ISSUE #114: Wrong devfile picked when Nodejs express and Java Maven are detected in a project
When analyzing a project containing Java Maven and Nodejs Express, alizer should pick the devfile related to nodejs because it has a framework detected in it. This does not happen when using the staging Devfile registry at https://registry.stage.devfile.io and https://github.com/quarkusio/quarkus-super-heroes/tree/main/ui-super-heroes

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #114 https://github.com/redhat-developer/alizer/issues/117

This patch contains some improvements to solve the issue faced with the `quarkus super heroes ui` app.

- it checks devDependencies to search for the framework
- it accepts multiple components detection in same path (e.g java and typescript in the root)
- it adds a new method to retrieve a list of devfiles for the project
- JavaScript and TypeScript are considered equivalent when picking a devfile. So a ts devfile is valid for a js project and viceversa.

The ouput of the `quarkus super heroes ui` app is 

```
[
        {
                "Name": "java-maven",
                "Language": "Java",
                "ProjectType": "Maven",
                "Tags": [
                        "Java",
                        "Maven"
                ]
        },
        {
                "Name": "java-openliberty",
                "Language": "Java",
                "ProjectType": "Open Liberty",
                "Tags": [
                        "Java",
                        "Maven"
                ]
        },
        {
                "Name": "java-websphereliberty",
                "Language": "Java",
                "ProjectType": "WebSphere Liberty",
                "Tags": [
                        "Java",
                        "Maven",
                        "WebSphere Liberty"
                ]
        },
        {
                "Name": "java-wildfly-bootable-jar",
                "Language": "Java",
                "ProjectType": "WildFly",
                "Tags": [
                        "RHEL8",
                        "Java",
                        "OpenJDK",
                        "Maven",
                        "WildFly",
                        "Microprofile",
                        "WildFly Bootable"
                ]
        },
        {
                "Name": "nodejs",
                "Language": "JavaScript",
                "ProjectType": "Node.js",
                "Tags": [
                        "Node.js",
                        "Express",
                        "ubi8"
                ]
        },
        {
                "Name": "nodejs-angular",
                "Language": "TypeScript",
                "ProjectType": "Angular",
                "Tags": [
                        "Node.js",
                        "Angular"
                ]
        }
]
```

### CODE REVIEW

**Comment by lstocchi** [comment]
@kadel tried to implement the "multiple devfiles detection". First thing i noticed is that in some cases it could return several devfiles which is not nice. E.g for a maven/angular application i got

```
[
	{
		"Name": "java-maven",
		"Language": "Java",
		"ProjectType": "Maven",
		"Tags": [
			"Java",
			"Maven"
		]
	},
	{
		"Name": "java-openliberty",
		"Language": "Java",
		"ProjectType": "Open Liberty",
		"Tags": [
			"Java",
			"Maven"
		]
	},
	{
		"Name": "java-websphereliberty",
		"Language": "Java",
		"ProjectType": "WebSphere Liberty",
		"Tags": [
			"Java",
			"Maven",
			"WebSphere Liberty"
		]
	},
	{
		"Name": "java-wildfly-bootable-jar",
		"Language": "Java",
		"ProjectType": "WildFly",
		"Tags": [
			"RHEL8",
			"Java",
			"OpenJDK",
			"Maven",
			"WildFly",
			"Microprofile",
			"WildFly Bootable"
		]
	},
	{
		"Name": "nodejs-angular",
		"Language": "TypeScript",
		"ProjectType": "Angular",
		"Tags": [
			"Node.js",
			"Angular"
		]
	}
]
```

The outcome is correct as every java devfile contains the tag `Maven` in them but... does it look fine for you? My concern is that the final user would see this list and would start thinking "WTF is it asking me if i want to use a wildfly devfile?" ... :laughing: 
So do you want me to restrict the list somehow? E.g i would pick the devfiles `java-maven` and `java-openliberty` because we cover 100% of their tags so it's most likely that they are more relevant to the project? WDYT?

File: go/pkg/apis/recognizer/devfile_recognizer.go [hunk 1]
```diff
@@ -19,44 +19,95 @@ import (
...
+	}
+	return []int{}, errors.New("no valid devfile found by using those languages")
+}
+
+func SelectDevFileUsingLanguagesFromTypes(languages []model.Language, devFileTypes []model.DevFileType) (int, error) {
+	devFilesIndexes, err := SelectDevFilesUsingLanguagesFromTypes(languages, devFileTypes)
+	if err != nil {
+		return -1, err
+	}
+	return devFilesIndexes[0], nil
```
Comment (jeffmaury) [inline go/pkg/apis/recognizer/devfile_recognizer.go:91]: Don't we face an index out of bounds if array is empty ?

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by lstocchi** [comment]
@kadel now alizer provides the functions that return a list of devfiles. You just need to use the proper function (just add an `s` to devfile keyword.)
E.g
```
SelectDevFileFromRegistry -> SelectDevFilesFromRegistry
SelectDevFileUsingLanguagesFromTypes -> SelectDevFilesUsingLanguagesFromTypes
SelectDevFileFromTypes -> SelectDevFilesFromTypes
```

**Pull request merged**